src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,511 @@
+        }
+
+        internal bool DtrEnable
+        {
+            get
+            {
+                int status = Interop.Sys.TerminalGetDtr(_handle);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+                return status == 1;
+            }
+            set
+            {
+                if (Interop.Sys.TerminalSetDtr(_handle, value ? 1 : 0) != 0)
+                {
+                    throw new IOException();
+                }
+            }
+        }
+
+        internal bool RtsEnable
+        {
+            get
+            {
+                int status = Interop.Sys.TerminalGetRts(_handle);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+                return status == 1;
+            }
+            set
+            {
+                if (Interop.Sys.TerminalSetRts(_handle, value ? 1 : 0) != 0)
+                {
+                    throw new IOException();
+                }
+            }
+        }
+
+        internal Handshake Handshake
+        {
+            set
+            {
+                Debug.Assert(!(value < Handshake.None || value > Handshake.RequestToSendXOnXOff),
+                    "An invalid value was passed to Handshake");
+
+                if (value != _handshake)
+                {
+
+                }
+            }
+        }
+
+        internal int DataBits
+        {
+            set
+            {
+                Debug.Assert(!(value < MinDataBits || value > MaxDataBits), "An invalid value was passed to DataBits");
+                _dataBits = value;
+            }
+        }
+
+        internal Parity Parity
+        {
+            set
+            {
+                Debug.Assert(!(value < Parity.None || value > Parity.Space), "An invalid value was passed to Parity");
+
+                if (value != _parity)
+                {
+
+                }
+            }
+        }
+
+        internal StopBits StopBits
+        {
+            set
+            {
+                Debug.Assert(!(value < StopBits.One || value > StopBits.OnePointFive), "An invalid value was passed to StopBits");
+
+            }
+        }
+
+        internal bool DiscardNull
+        {
+            set
+            {
+                // Ignore.
+            }
+        }
+
+        internal byte ParityReplace
+        {
+            set
+            {
+                // Ignore.
+            }
+        }
+
+       // Uses Win32 method to dump out the receive buffer; analagous to MSComm's "InBufferCount = 0"
+        internal void DiscardInBuffer()
+        {
+        }
+
+        // Uses Win32 method to dump out the xmit buffer; analagous to MSComm's "OutBufferCount = 0"
+        internal void DiscardOutBuffer()
+        {
+        }
+
+       internal void SetBufferSizes(int readBufferSize, int writeBufferSize)
+        {
+            if (_handle == null) throw new IOException();
```

Comment:
That should be InvalidOperationException, or something else, not an actual IOException.